### PR TITLE
Added a README.md file

### PR DESCRIPTION
**Pull request by AGalabov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,196 @@
...
+          type: string
+          example: '1212121'
+        name:
+          type: string
+          example: 'John Doe'
+        age:
+          type: number
+          example: 42
+      required:
+        - userId
```
Comment (georgyangelov) [inline README.md:52]: `id`

File: README.md [hunk 2]
```diff
@@ -1,2 +1,196 @@
...
+```
+
+Obviously there is a lot of duplication when defining the zod schema and the schema for Open API. And let's face it - writing code with intellisense is much easier than writing YAML.
+
+### The solution
+
+With z
+
+```ts
+extendZodWithOpenApi(z);
```
Comment (georgyangelov) [inline README.md:89]: Please show the imports as well and mention that this should be done only once in a common-entrypoint file.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,196 @@
...
+// If needed to you can always use yaml.stringify()
+```
+
+## How it works
+
+### Technologies
+
+- [Typescript](https://www.typescriptlang.org/) first.
+- Schema validation — [Zod 3.x](https://github.com/colinhacks/zod).
+- Documenting — [OpenAPI 3.x](https://github.com/metadevpro/openapi3-ts).
```
Comment (georgyangelov) [inline README.md:144]: What is `Documenting`?

File: README.md [hunk 4]
```diff
@@ -1,2 +1,196 @@
...
+
+In a main file somewhere in your file system:
+
+```ts
+import { z } from 'zod';
+import {
+  extendZodWithOpenApi,
+  OpenAPIGenerator,
+  OpenAPIRegistry,
+} from 'zod-t-openapi';
```
Comment (georgyangelov) [inline README.md:174]: `@asteasolutions/zod-to-openapi`

File: README.md [hunk 5]
```diff
@@ -1,2 +1,196 @@
...
+      mediaType: 'application/json',
+      schema: UserSchema.openapi({
+        description: 'Object with user data.',
+      }),
+    },
+  },
+});
+
+const generator = new OpenAPIGenerator(rootRegistry.definitions);
+generator.generateDocument({ ...somBaseOpenApiConfig }); // this would return an object in the OpenAPI format
```
Comment (georgyangelov) [inline README.md:134]: `somBaseOpenApiConfig` -> `someBaseOpenApiConfig`

File: README.md [hunk 6]
```diff
@@ -1,2 +1,196 @@
...
+      in: header
+      name: id
+      schema:
+        type: string
+      required: true
+      description: The user id to look for
+      allowReserved: true
+      example: '1212121'
+# ... Some other stuff in between
+paths:
```
Comment (georgyangelov) [inline README.md:66]: It's a bit confusing to show TS code and then show more than what it generates - users might get the wrong idea. I suggest splitting the example with the schemas and the one with the paths

File: README.md [hunk 7]
```diff
@@ -1,2 +1,257 @@
...
+  name: id
+  schema:
+    type: string
+    example: '1212121'
+  required: true
+```
+
+The result would be an object like `{ components: { parameters: { UserId: {...} } } }`. The key for the object is the value of the first argument passed to `.registerParameter` (in this case - `UserId`).
+
+The resulting schema can then be referenced by using `$ref: #/components/parameters/User` in an existing OpenAPI JSON.
```
Comment (blagowestas) [inline README.md:151]: Maybe this should be `$ref: #/components/parameters/UserId`?

File: README.md [hunk 8]
```diff
@@ -1,2 +1,257 @@
...
+
+An OpenAPI parameter (query/path/header) should be registered using the `registerParameter` method of an `OpenAPIRegistry` instance.
+
+```ts
+const UserIdSchema = registry.registerParameter(
+  'UserId',
+  z.string().openapi({
+    param: {
+      name: 'id',
+      in: 'path',
```
Comment (blagowestas) [inline README.md:128]: Maybe `path` can be `example-path`

File: README.md [hunk 9]
```diff
@@ -1,2 +1,257 @@
...
+  info: {
+    version: '1.0.0',
+    title: 'My API',
+    description: 'This is the API',
+  },
+  servers: [{ url: 'v1' }],
+});
+```
+
+#### Registering a path
```
Comment (blagowestas) [inline README.md:169]: Is this the right place for this paragraph?

File: README.md (hunk 8 above)
Comment (AGalabov) [reply README.md:128]: The `in` key is openapi specific - `'query' | 'header' | 'path' | 'cookie'`

File: README.md [hunk 10]
```diff
@@ -1,2 +1,257 @@
 # Zod to OpenAPI
 
+A library that uses zod schemas to generate an Open API Swagger documentation.
+
+1. [Purpose](#purpose)
+2. [Usage](#usage)
+   1. [Installation](#installation)
```
[the remark this answers is not in the document]
Comment (AGalabov) [reply README.md:7]: Not really - it's how the Github parser decides to "render" it.

**Review by blagowestas — APPROVED** [review]

**Pull request merged**